Backup: init 0.2.0-alpha cycle

samiff opened the pull request:
#### Changes proposed in this Pull Request:

* After testing tooling for 0.1.0-beta [release](https://github.com/Automattic/jetpack-backup-plugin/releases), init new release cycle
* Add `production-exclude` for `src/js` in `.gitattributes` as JS/CSS files are served in `/build`
* Corrects an ignored filename in `.gitattributes`

#### Jetpack product discussion

* N/A

#### Does this pull request change what data or activity we track or use?

* No.

#### Testing instructions:

* Similar to JP release cycle, review changes.

samiff committed:
Backup: init 0.2.0-alpha cycle

Ran: tools/project-version.sh -u 0.2.0-alpha plugins/backup

samiff committed:
Add changelog

samiff committed:
Fix a filename in gitattributes

samiff committed:
Exclude src/js in gitattributes for production

samiff commented:
@anomiex I was able to test the .zip from the mirror repo release and that worked ok for me 👍 

I'm thinking that `src/js` can be excluded from the mirror repo (tests fine for me that way), but thought I'd double check: 9f5c7ca

anomiex commented:
> I'm thinking that `src/js` can be excluded from the mirror repo (tests fine for me that way), but thought I'd double check: [9f5c7ca](https://github.com/Automattic/jetpack/commit/9f5c7ca25f3248293b5656e55e8ab8c955925aa3)

It probably can, I don't see any use of `Assets::get_file_url_for_environment()` or the like that tries to load the sources directly for a debug mode. I don't know if there's any benefit to humans in including the sources in the plugin zip.

anomiex commented:
You've got some merge conflicts now that need to be resolved.

samiff committed:
Merge branch 'master' into update/backup-init-0_2_0-alpha

samiff commented:
@anomiex 

> Error: Version mismatch, expected 0.1.1-alpha but found 0.2.0-alpha! You might use `tools/project-version.sh -u 0.1.1-alpha plugins/backup` to fix this.

Say I wanted the 0.2.0-alpha version applied, what would be the way around that?

anomiex commented:
Create a change entry file with significance set to "minor". It can be one without an actual entry text if you want.

anomiex commented on projects/plugins/backup/changelog/update-backup-init-0_2_0-alpha:
```diff
@@ -0,0 +1,5 @@
+Significance: patch
```
i.e.
```suggestion
Significance: minor
```

samiff replied:
That makes sense, thanks!

samiff committed:
Update changelog significance

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

samiff committed:
Update update-backup-init-0_2_0-alpha

samiff committed:
Merge branch 'master' into update/backup-init-0_2_0-alpha

samiff committed:
Fix script name

samiff committed:
Update pkg name as to not conflict with backup pkg

The Backup plugin will need to require the `automattic/jetpack-backup`
package in the future, so this avoids the following conflict:

Root package 'automattic/jetpack-backup' cannot require itself in its
composer.json

samiff committed:
Update lock file

samiff commented:
@anomiex Could you help me understand the failing `Linting / Lock files are up to date` check please. I did run & [commit](a3eb81d) the results of `tools/composer-update-monorepo.sh --root-reqs "projects/plugins/backup"`, but not sure why it's not matching up at the moment.

anomiex commented on projects/plugins/backup/.gitattributes:
```diff
@@ -17,7 +17,8 @@ composer.lock                               production-exclude
 package.json                                production-exclude
 phpunit.xml.dist                            production-exclude
-readme.md                                   production-exclude
+README.md                                   production-exclude
```
Seems odd to be excluding README.md from the mirror. We probably want it to show up at https://github.com/Automattic/jetpack-backup-plugin#readme.

Look at the unfriendly display at https://github.com/Automattic/jetpack-production#readme for a counterexample. We should fix (in a different PR, of course).

samiff replied:
Thanks Brad, changing L213 back seemed to have worked, not entirely sure what got mixed up there.

Regarding not showing `README.md` in the mirror repo, I was just following Jetack's lead there. If I understand correctly we'll need the `.txt` with the stable tag for releasing on WPorg, so it might be simpler to just maintain `readme.txt` in the mirror repo? @kraftbj any thoughts on this?

anomiex replied:
Yeah, we need readme.txt for wporg. But GH displays readme.txt as plain text, not markdown.

anomiex reviewed (commented):
The problem seems to be [your change to line 213 in a3eb81da](https://github.com/Automattic/jetpack/pull/20084/commits/a3eb81daca1b4a89d9806412f3266a5508a566c8#diff-f808bdbb4edff87e513237afc5cbd6c683ffc94a5c8fa9e7eeb58ea1223fec46R213), it wants that changed back. See [the diff in the log](https://github.com/Automattic/jetpack/pull/20084/checks?check_run_id=2852671025#step:11:171).

Did you have uncommitted changes to projects/packages/connection/composer.json when you ran the command or something?

samiff committed:
Fix lock file check
